compiler/rustc_mir_transform/src/function_item_references.rs

```diff
@@ -161,7 +161,7 @@ impl<'tcx> FunctionItemRefChecker<'_, 'tcx> {
             .as_ref()
             .assert_crate_local()
             .lint_root;
-        let fn_sig = self.tcx.fn_sig(fn_id);
+        let fn_sig = self.tcx.fn_sig(fn_id).skip_binder();
```

Comment:
i feel like that should use `subst(self.tcx, fn_substs)`? also, why are we manually printing the fn sig here? can you add a `FIXME: use existing printing routines to print the function signature?`? or impl that yourself, whatever you prefer.

Reply:
Okay, I looked into the `FIXME` and I think we should leave this one alone. It's printing the `fn_sig` to appear on the rhs of a cast, as can be seen in the [ui test output](https://github.com/rust-lang/rust/blob/master/tests/ui/lint/function-item-references.stderr). I looked at `FnSig::print` / `ty::print::PrettyPrinter`, but that looks like it does something different (i.e., writes out the parameter types etc)

Maybe no need to put the `FIXME` here? (unless you know of another print routine that already does this)